packages/fx-core/src/component/generator/generatorAction.ts

```diff
@@ -77,6 +79,13 @@ export const fetchTemplateZipFromSourceCodeAction: GeneratorAction = {
   },
 };
 
+export const downloadDirectoryAction: GeneratorAction = {
+  name: GeneratorActionName.DownloadDirectory,
+  run: async (context: GeneratorContext) => {
+    await downloadDirectory(context.url!, context.destination);
```

Comment:
Could you add null-check for context.url? We should avoid using non-null assertion as possible.